packages/core/src/index.ts

```diff
@@ -62,14 +63,23 @@ const inputValidationError = (inputNode: HTMLInputElement): null | string => {
   return null;
 };
 
-const readVcAttributes = (node: HTMLElement, eventType: string) => {
+const readVcAttributes = <T>(
+  node: HTMLElement,
+  eventType: string
+): {
+  stepId: StepData<T>["stepId"];
+  journeyId: StepData<T>["journeyId"];
+  escalate: StepData<T>["escalate"];
+  end: StepData<T>["end"];
+  payload: StepData<T>["payload"];
+} | null => {
   const stepId = node.getAttribute(`vc-${eventType}-stepid`);
   if (!stepId) {
     return null;
   }
   return {
     stepId,
-    journeyId: node.getAttribute(`vc-${eventType}-journeyid`),
+    journeyId: node.getAttribute(`vc-${eventType}-journeyid`) ?? undefined,
```

Comment:
returning null messes with the updateStep type, setting to undefined enables the destructuring assignment i'm using

Reply:
That makes sense!